factory/dataComponents.go

```diff
@@ -183,5 +183,13 @@ func (cc *dataComponents) Close() error {
 		}
 	}
 
+	if !check.IfNil(cc.datapool) && !check.IfNil(cc.datapool.PeerAuthentications()) {
```

Comment:
this is correct :+1: 

Please change the behavior of this function: each time a new error is found, it will immediately return from this function. This is not great as some others components might remain opened, increasing the pressure on the RAM usage.
Instead, let's log with error and store only the last error (if we encounter one) and finally return that error on L194.